### PR TITLE
Obelisk-B-Gone PART 1

### PR DESCRIPTION
**Pull request by Mycah142**
## About The Pull Request
This is Part One of my goal to make Obelisks destructible in meaningful and interesting ways.

This is the Deconstruction Litany, added to the Acolyte tab of the ritual book that allows any Acolyte to deconstruct anything they can construct with litanies back to their base materials.

More interesting/nonchurch ways will be done in the coming future, soon(tm).

Also a minor fix related to the list generation, the Construction.dm contained a duplicated build_path for both the biogenerator console and bioreactor console. This has been corrected, as it fixed my own list generation.

An absolute special thank you to both Handyman and Kirov for their infinite patience with me on this. Without them I never would've gotten this done.

## Why It's Good For The Game
Obelisks are currently indestructible to anything but C4 or a Hatton. This at least opens up a peaceful avenue to allow the church to remove obelisks. Also allows them to deconstruct other Litany Constructs, like biomatter canisters.

## Testing
[dreamseeker_B2KiBtkznR.webm](https://user-images.githubusercontent.com/11076040/205204583-92c85f8d-8f59-4421-b384-4ec0efcdf484.webm)

## Changelog
:cl:
add: Uproot: Deconstruction Litany
fix: Fixed build_path of Bioreactor Metrics console
/:cl:

### CODE REVIEW

File: code/modules/core_implant/cruciform/rituals/construction.dm [hunk 1]
```diff
@@ -74,6 +85,52 @@ GLOBAL_LIST_INIT(nt_blueprints, init_nt_blueprints())
...
 
+/datum/ritual/cruciform/priest/acolyte/deconstruction
+	name = "Uproot"
+	phrase = "Dominus dedit, Dominus abstulit."
+	desc = "Mistakes are to be a lesson, but first we must correct it by deconstructing its form."
+	power = 40
+
+/datum/ritual/cruciform/priest/acolyte/deconstruction/perform(mob/living/carbon/human/user, obj/item/implant/core_implant/C, list/targets)
+	if(!GLOB.nt_constructs) //Makes sure the list we curated earlier actually exists
+		fail("You have no idea what constitutes a church construct.",user,C,targets)
```
Comment (DiaFRAME444) [inline code/modules/core_implant/cruciform/rituals/construction.dm:96]: Add a comment saying to report this occurence to the discord.

File: code/modules/core_implant/cruciform/rituals/construction.dm [hunk 2]
```diff
@@ -74,6 +85,52 @@ GLOBAL_LIST_INIT(nt_blueprints, init_nt_blueprints())
 	var/build_path = blueprint.build_path
 	new build_path(target_turf)
 
+/datum/ritual/cruciform/priest/acolyte/deconstruction
+	name = "Uproot"
+	phrase = "Dominus dedit, Dominus abstulit."
+	desc = "Mistakes are to be a lesson, but first we must correct it by deconstructing its form."
+	power = 40
```
Comment (DiaFRAME444) [inline code/modules/core_implant/cruciform/rituals/construction.dm:92]: Why is the power cost of this so high? I hate being cruciform energy cucked...

Comment (Mycah142) [reply code/modules/core_implant/cruciform/rituals/construction.dm:92]: Same as Manifestation, the construction litany.

Litany powers will be looked at later as a whole.

**Review by SirRichardFrancis — APPROVED** [review]

**Review by Pink-Chink — APPROVED** [review]
Nice video demo

File: code/modules/core_implant/cruciform/rituals/construction.dm [hunk 3]
```diff
@@ -74,6 +85,52 @@ GLOBAL_LIST_INIT(nt_blueprints, init_nt_blueprints())
...
+	for(reclaimed in target_turf) 
+		if(reclaimed.type in GLOB.nt_constructs)
+			loot = GLOB.nt_constructs[reclaimed.type]
+			break
+
+	if(isnull(loot))
+		fail("There is no mistake to remove here.",user,C,targets)
+		return
+
+	user.visible_message(SPAN_NOTICE("You see as [user] places one hand upon their cruciform, and the other stretched forward."),SPAN_NOTICE("You take back what is, returning it to what was."))
```
Comment (Firefox13) [inline code/modules/core_implant/cruciform/rituals/construction.dm:113]:
```suggestion
	user.visible_message(SPAN_NOTICE("[user] places one hand on their chest, and the other stretched forward."),SPAN_NOTICE("You take back what is, returning it to what was."))
```
I still think it's some poor wording

**Review by Firefox13 — COMMENTED** [review]
There is a feedback to c4ing an obelisk (explosion) - this also needs some evidence.

**Comment by Mycah142** [comment]
> There is a feedback to c4ing an obelisk (explosion) - this also needs some evidence.

What do you mean? C4 hasn't changed. All changes I made are contained entirely in the new litany, how C4 qdels Obelisks hasn't changed.

**Comment by Firefox13** [comment]
There is evidence after you C4 an obelisk - hear able explosion and some broken tiles afterwards. Using a litany should give a warning to Church members akin to call for help litany

**Comment by Mycah142** [comment]
> There is evidence after you C4 an obelisk - hear able explosion and some broken tiles afterwards. Using a litany should give a warning to Church members akin to call for help litany

I planned on such for Part 2, though not on the litany, as that would be a specific obelisk reaction and not part of the litany, as the Litany is not heretical and only accessible by Acolytes, and thus "approved".

In Part 2 I'm adding not-church ways to destroy the obelisk that will send an Entreaty.

**Comment by Humonitarian** [comment]
> the Litany is not heretical and only accessible by Acolytes, and thus "approved"

I do agree with Firefox here, there's no hurt to making it be announced so the Acolyte has to declare their intentions.
Will be useful if we ever get the Heretic antag, or in general if some inquisitor starts messing with the church.

I think it is more than enough if the simple act is announced without naming the acolyte in question, though.

**Comment by Mycah142** [comment]
> > the Litany is not heretical and only accessible by Acolytes, and thus "approved"
> 
> I do agree with Firefox here, there's no hurt to making it be announced so the Acolyte has to declare their intentions. Will be useful if we ever get the Heretic antag, or in general if some inquisitor starts messing with the church.
> 
> I think it is more than enough if the simple act is announced without naming the acolyte in question, though.

This would mean making a new proc or adding snowflake reaction to obelisks alone.

Unless I make it send an Entreaty on every object it removes, as that would make sense to have on a litany that removes more than just obelisks. The video even shows it removes doors too.

**Comment by DiaFRAME444** [comment]
Praise be

**Pull request merged**